### ISSUE TITLE
[Bug Report] App crashes when searching for school

### ISSUE THREAD

**Issue by ManuelKlaer** [SapuSeven/BetterUntis#224]
**Description**

I just updated to the new version of BetterUntis and was hoping I could finally use it again. However, the app keeps crashing when I search for a school. _(I constantly end up back on the home screen with no error message in the app itself. Therefore, I used logcat and got the error message this way.)_

_(Translated with DeepL)_

<details>
  <summary><b>Logs</b></summary>

```
12-09 18:31:57.751  9854  9854 E BetterUntis: Application crashed!
12-09 18:31:57.751  9854  9854 E BetterUntis: v7.h: Unexpected JSON token at offset 345: Unexpected 'null' value instead of string literal
12-09 18:31:57.751  9854  9854 E BetterUntis: JSON input: .....elden","mobileServiceUrl":null},{"server":"nete.webuntis.com.....
12-09 18:31:57.751  9854  9854 E BetterUntis:   at u7.j.d(Unknown Source:35)
12-09 18:31:57.751  9854  9854 E BetterUntis:   at u7.j.e(Unknown Source:34)
12-09 18:31:57.751  9854  9854 E BetterUntis:   at v7.a.o(Unknown Source:9)
12-09 18:31:57.751  9854  9854 E BetterUntis:   at v7.a.p(Unknown Source:6)
12-09 18:31:57.751  9854  9854 E BetterUntis:   at v7.a.m(:2)
12-09 18:31:57.751  9854  9854 E BetterUntis:   at v7.s.A(:2)
12-09 18:31:57.751  9854  9854 E BetterUntis:   at s7.a.p(Unknown Source:5)
12-09 18:31:57.751  9854  9854 E BetterUntis:   at com.sapuseven.untis.models.UntisSchoolInfo$$serializer.deserialize(Unknown Source:118)
12-09 18:31:57.751  9854  9854 E BetterUntis:   at com.sapuseven.untis.models.UntisSchoolInfo$$serializer.deserialize(Unknown Source:0)
12-09 18:31:57.751  9854  9854 E BetterUntis:   at u7.j.i(:11)
12-09 18:31:57.751  9854  9854 E BetterUntis:   at v7.s.i(Unknown Source:5)
12-09 18:31:57.751  9854  9854 E BetterUntis:   at s7.a.o(:1)
12-09 18:31:57.751  9854  9854 E BetterUntis:   at s7.c$a.a(Unknown Source:1)
12-09 18:31:57.751  9854  9854 E BetterUntis:   at t7.e0.h(Unknown Source:17)
12-09 18:31:57.751  9854  9854 E BetterUntis:   at t7.a.f(:3)
12-09 18:31:57.751  9854  9854 E BetterUntis:   at t7.a.deserialize(Unknown Source:6)
12-09 18:31:57.751  9854  9854 E BetterUntis:   at u7.j.i(:11)
12-09 18:31:57.751  9854  9854 E BetterUntis:   at v7.s.i(Unknown Source:5)
12-09 18:31:57.751  9854  9854 E BetterUntis:   at s7.a.o(:1)
12-09 18:31:57.751  9854  9854 E BetterUntis:   at com.sapuseven.untis.models.untis.response.SchoolSearchResult$$serializer.deserialize(Unknown Source:63)
12-09 18:31:57.751  9854  9854 E BetterUntis:   at com.sapuseven.untis.models.untis.response.SchoolSearchResult$$serializer.deserialize(Unknown Source:0)
12-09 18:31:57.751  9854  9854 E BetterUntis:   at u7.j.i(:11)
12-09 18:31:57.751  9854  9854 E BetterUntis:   at v7.s.i(Unknown Source:5)
12-09 18:31:57.751  9854  9854 E BetterUntis:   at s7.a.E(:1)
12-09 18:31:57.751  9854  9854 E BetterUntis:   at com.sapuseven.untis.models.untis.response.SchoolSearchResponse$$serializer.deserialize(Unknown Source:75)
12-09 18:31:57.751  9854  9854 E BetterUntis:   at com.sapuseven.untis.models.untis.response.SchoolSearchResponse$$serializer.deserialize(Unknown Source:0)
12-09 18:31:57.751  9854  9854 E BetterUntis:   at u7.j.i(:11)
12-09 18:31:57.751  9854  9854 E BetterUntis:   at v7.s.i(Unknown Source:5)
12-09 18:31:57.751  9854  9854 E BetterUntis:   at u7.a.b(Unknown Source:21)
12-09 18:31:57.751  9854  9854 E BetterUntis:   at m3.r.e(:27)
12-09 18:31:57.751  9854  9854 E BetterUntis:   at r4.a.l(:4)
12-09 18:31:57.751  9854  9854 E BetterUntis:   at k7.f0.run(:2)
12-09 18:31:57.751  9854  9854 E BetterUntis:   at android.os.Handler.handleCallback(Handler.java:938)
12-09 18:31:57.751  9854  9854 E BetterUntis:   at android.os.Handler.dispatchMessage(Handler.java:99)
12-09 18:31:57.751  9854  9854 E BetterUntis:   at android.os.Looper.loop(Looper.java:233)
12-09 18:31:57.751  9854  9854 E BetterUntis:   at android.app.ActivityThread.main(ActivityThread.java:8030)
12-09 18:31:57.751  9854  9854 E BetterUntis:   at java.lang.reflect.Method.invoke(Native Method)
12-09 18:31:57.751  9854  9854 E BetterUntis:   at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:631)
12-09 18:31:57.751  9854  9854 E BetterUntis:   at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:978)
12-09 18:31:57.751  9854  9854 W c       : Unexpected JSON token at offset 345: Unexpected 'null' value instead of string literal
12-09 18:31:57.751  9854  9854 W c       : JSON input: .....elden","mobileServiceUrl":null},{"server":"nete.webuntis.com.....
12-09 18:31:57.757  9854  9854 E BetterUntis: Application crashed!
12-09 18:31:57.757  9854  9854 E BetterUntis: v7.h: Unexpected JSON token at offset 345: Unexpected 'null' value instead of string literal
12-09 18:31:57.757  9854  9854 E BetterUntis: JSON input: .....elden","mobileServiceUrl":null},{"server":"nete.webuntis.com.....
12-09 18:31:57.757  9854  9854 E BetterUntis:   at u7.j.d(Unknown Source:35)
12-09 18:31:57.757  9854  9854 E BetterUntis:   at u7.j.e(Unknown Source:34)
12-09 18:31:57.757  9854  9854 E BetterUntis:   at v7.a.o(Unknown Source:9)
12-09 18:31:57.757  9854  9854 E BetterUntis:   at v7.a.p(Unknown Source:6)
12-09 18:31:57.757  9854  9854 E BetterUntis:   at v7.a.m(:2)
12-09 18:31:57.757  9854  9854 E BetterUntis:   at v7.s.A(:2)
12-09 18:31:57.757  9854  9854 E BetterUntis:   at s7.a.p(Unknown Source:5)
12-09 18:31:57.757  9854  9854 E BetterUntis:   at com.sapuseven.untis.models.UntisSchoolInfo$$serializer.deserialize(Unknown Source:118)
12-09 18:31:57.757  9854  9854 E BetterUntis:   at com.sapuseven.untis.models.UntisSchoolInfo$$serializer.deserialize(Unknown Source:0)
12-09 18:31:57.757  9854  9854 E BetterUntis:   at u7.j.i(:11)
12-09 18:31:57.757  9854  9854 E BetterUntis:   at v7.s.i(Unknown Source:5)
12-09 18:31:57.757  9854  9854 E BetterUntis:   at s7.a.o(:1)
12-09 18:31:57.757  9854  9854 E BetterUntis:   at s7.c$a.a(Unknown Source:1)
12-09 18:31:57.757  9854  9854 E BetterUntis:   at t7.e0.h(Unknown Source:17)
12-09 18:31:57.757  9854  9854 E BetterUntis:   at t7.a.f(:3)
12-09 18:31:57.757  9854  9854 E BetterUntis:   at t7.a.deserialize(Unknown Source:6)
12-09 18:31:57.757  9854  9854 E BetterUntis:   at u7.j.i(:11)
12-09 18:31:57.757  9854  9854 E BetterUntis:   at v7.s.i(Unknown Source:5)
12-09 18:31:57.757  9854  9854 E BetterUntis:   at s7.a.o(:1)
12-09 18:31:57.757  9854  9854 E BetterUntis:   at com.sapuseven.untis.models.untis.response.SchoolSearchResult$$serializer.deserialize(Unknown Source:63)
12-09 18:31:57.757  9854  9854 E BetterUntis:   at com.sapuseven.untis.models.untis.response.SchoolSearchResult$$serializer.deserialize(Unknown Source:0)
12-09 18:31:57.757  9854  9854 E BetterUntis:   at u7.j.i(:11)
12-09 18:31:57.757  9854  9854 E BetterUntis:   at v7.s.i(Unknown Source:5)
12-09 18:31:57.757  9854  9854 E BetterUntis:   at s7.a.E(:1)
12-09 18:31:57.757  9854  9854 E BetterUntis:   at com.sapuseven.untis.models.untis.response.SchoolSearchResponse$$serializer.deserialize(Unknown Source:75)
12-09 18:31:57.757  9854  9854 E BetterUntis:   at com.sapuseven.untis.models.untis.response.SchoolSearchResponse$$serializer.deserialize(Unknown Source:0)
12-09 18:31:57.757  9854  9854 E BetterUntis:   at u7.j.i(:11)
12-09 18:31:57.757  9854  9854 E BetterUntis:   at v7.s.i(Unknown Source:5)
12-09 18:31:57.757  9854  9854 E BetterUntis:   at u7.a.b(Unknown Source:21)
12-09 18:31:57.757  9854  9854 E BetterUntis:   at m3.r.e(:27)
12-09 18:31:57.757  9854  9854 E BetterUntis:   at r4.a.l(:4)
12-09 18:31:57.757  9854  9854 E BetterUntis:   at k7.f0.run(:2)
12-09 18:31:57.757  9854  9854 E BetterUntis:   at android.os.Handler.handleCallback(Handler.java:938)
12-09 18:31:57.757  9854  9854 E BetterUntis:   at android.os.Handler.dispatchMessage(Handler.java:99)
12-09 18:31:57.757  9854  9854 E BetterUntis:   at android.os.Looper.loop(Looper.java:233)
12-09 18:31:57.757  9854  9854 E BetterUntis:   at android.app.ActivityThread.main(ActivityThread.java:8030)
12-09 18:31:57.757  9854  9854 E BetterUntis:   at java.lang.reflect.Method.invoke(Native Method)
12-09 18:31:57.757  9854  9854 E BetterUntis:   at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:631)
12-09 18:31:57.757  9854  9854 E BetterUntis:   at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:978)
12-09 18:31:57.757  9854  9854 W c       : Unexpected JSON token at offset 345: Unexpected 'null' value instead of string literal
12-09 18:31:57.757  9854  9854 W c       : JSON input: .....elden","mobileServiceUrl":null},{"server":"nete.webuntis.com.....
12-09 18:31:57.760  9854  9854 E BetterUntis: Application crashed!
12-09 18:31:57.760  9854  9854 E BetterUntis: v7.h: Unexpected JSON token at offset 345: Unexpected 'null' value instead of string literal
12-09 18:31:57.760  9854  9854 E BetterUntis: JSON input: .....elden","mobileServiceUrl":null},{"server":"nete.webuntis.com.....
12-09 18:31:57.760  9854  9854 E BetterUntis:   at u7.j.d(Unknown Source:35)
12-09 18:31:57.760  9854  9854 E BetterUntis:   at u7.j.e(Unknown Source:34)
12-09 18:31:57.760  9854  9854 E BetterUntis:   at v7.a.o(Unknown Source:9)
12-09 18:31:57.760  9854  9854 E BetterUntis:   at v7.a.p(Unknown Source:6)
12-09 18:31:57.760  9854  9854 E BetterUntis:   at v7.a.m(:2)
12-09 18:31:57.760  9854  9854 E BetterUntis:   at v7.s.A(:2)
12-09 18:31:57.760  9854  9854 E BetterUntis:   at s7.a.p(Unknown Source:5)
12-09 18:31:57.760  9854  9854 E BetterUntis:   at com.sapuseven.untis.models.UntisSchoolInfo$$serializer.deserialize(Unknown Source:118)
12-09 18:31:57.760  9854  9854 E BetterUntis:   at com.sapuseven.untis.models.UntisSchoolInfo$$serializer.deserialize(Unknown Source:0)
12-09 18:31:57.760  9854  9854 E BetterUntis:   at u7.j.i(:11)
12-09 18:31:57.760  9854  9854 E BetterUntis:   at v7.s.i(Unknown Source:5)
12-09 18:31:57.760  9854  9854 E BetterUntis:   at s7.a.o(:1)
12-09 18:31:57.760  9854  9854 E BetterUntis:   at s7.c$a.a(Unknown Source:1)
12-09 18:31:57.760  9854  9854 E BetterUntis:   at t7.e0.h(Unknown Source:17)
12-09 18:31:57.760  9854  9854 E BetterUntis:   at t7.a.f(:3)
12-09 18:31:57.760  9854  9854 E BetterUntis:   at t7.a.deserialize(Unknown Source:6)
12-09 18:31:57.760  9854  9854 E BetterUntis:   at u7.j.i(:11)
12-09 18:31:57.760  9854  9854 E BetterUntis:   at v7.s.i(Unknown Source:5)
12-09 18:31:57.760  9854  9854 E BetterUntis:   at s7.a.o(:1)
12-09 18:31:57.760  9854  9854 E BetterUntis:   at com.sapuseven.untis.models.untis.response.SchoolSearchResult$$serializer.deserialize(Unknown Source:63)
12-09 18:31:57.760  9854  9854 E BetterUntis:   at com.sapuseven.untis.models.untis.response.SchoolSearchResult$$serializer.deserialize(Unknown Source:0)
12-09 18:31:57.760  9854  9854 E BetterUntis:   at u7.j.i(:11)
12-09 18:31:57.760  9854  9854 E BetterUntis:   at v7.s.i(Unknown Source:5)
12-09 18:31:57.760  9854  9854 E BetterUntis:   at s7.a.E(:1)
12-09 18:31:57.760  9854  9854 E BetterUntis:   at com.sapuseven.untis.models.untis.response.SchoolSearchResponse$$serializer.deserialize(Unknown Source:75)
12-09 18:31:57.760  9854  9854 E BetterUntis:   at com.sapuseven.untis.models.untis.response.SchoolSearchResponse$$serializer.deserialize(Unknown Source:0)
12-09 18:31:57.760  9854  9854 E BetterUntis:   at u7.j.i(:11)
12-09 18:31:57.760  9854  9854 E BetterUntis:   at v7.s.i(Unknown Source:5)
12-09 18:31:57.760  9854  9854 E BetterUntis:   at u7.a.b(Unknown Source:21)
12-09 18:31:57.760  9854  9854 E BetterUntis:   at m3.r.e(:27)
12-09 18:31:57.760  9854  9854 E BetterUntis:   at r4.a.l(:4)
12-09 18:31:57.760  9854  9854 E BetterUntis:   at k7.f0.run(:2)
12-09 18:31:57.760  9854  9854 E BetterUntis:   at android.os.Handler.handleCallback(Handler.java:938)
12-09 18:31:57.760  9854  9854 E BetterUntis:   at android.os.Handler.dispatchMessage(Handler.java:99)
12-09 18:31:57.760  9854  9854 E BetterUntis:   at android.os.Looper.loop(Looper.java:233)
12-09 18:31:57.760  9854  9854 E BetterUntis:   at android.app.ActivityThread.main(ActivityThread.java:8030)
12-09 18:31:57.760  9854  9854 E BetterUntis:   at java.lang.reflect.Method.invoke(Native Method)
12-09 18:31:57.760  9854  9854 E BetterUntis:   at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:631)
12-09 18:31:57.760  9854  9854 E BetterUntis:   at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:978)
12-09 18:31:57.760  9854  9854 W c       : Unexpected JSON token at offset 345: Unexpected 'null' value instead of string literal
12-09 18:31:57.760  9854  9854 W c       : JSON input: .....elden","mobileServiceUrl":null},{"server":"nete.webuntis.com.....
12-09 18:31:57.763  9854  9854 E AndroidRuntime: FATAL EXCEPTION: main
12-09 18:31:57.763  9854  9854 E AndroidRuntime: Process: com.sapuseven.untis, PID: 9854
12-09 18:31:57.763  9854  9854 E AndroidRuntime: v7.h: Unexpected JSON token at offset 345: Unexpected 'null' value instead of string literal
12-09 18:31:57.763  9854  9854 E AndroidRuntime: JSON input: .....elden","mobileServiceUrl":null},{"server":"nete.webuntis.com.....
12-09 18:31:57.763  9854  9854 E AndroidRuntime:        at u7.j.d(Unknown Source:35)
12-09 18:31:57.763  9854  9854 E AndroidRuntime:        at u7.j.e(Unknown Source:34)
12-09 18:31:57.763  9854  9854 E AndroidRuntime:        at v7.a.o(Unknown Source:9)
12-09 18:31:57.763  9854  9854 E AndroidRuntime:        at v7.a.p(Unknown Source:6)
12-09 18:31:57.763  9854  9854 E AndroidRuntime:        at v7.a.m(:2)
12-09 18:31:57.763  9854  9854 E AndroidRuntime:        at v7.s.A(:2)
12-09 18:31:57.763  9854  9854 E AndroidRuntime:        at s7.a.p(Unknown Source:5)
12-09 18:31:57.763  9854  9854 E AndroidRuntime:        at com.sapuseven.untis.models.UntisSchoolInfo$$serializer.deserialize(Unknown Source:118)
12-09 18:31:57.763  9854  9854 E AndroidRuntime:        at com.sapuseven.untis.models.UntisSchoolInfo$$serializer.deserialize(Unknown Source:0)
12-09 18:31:57.763  9854  9854 E AndroidRuntime:        at u7.j.i(:11)
12-09 18:31:57.763  9854  9854 E AndroidRuntime:        at v7.s.i(Unknown Source:5)
12-09 18:31:57.763  9854  9854 E AndroidRuntime:        at s7.a.o(:1)
12-09 18:31:57.763  9854  9854 E AndroidRuntime:        at s7.c$a.a(Unknown Source:1)
12-09 18:31:57.763  9854  9854 E AndroidRuntime:        at t7.e0.h(Unknown Source:17)
12-09 18:31:57.763  9854  9854 E AndroidRuntime:        at t7.a.f(:3)
12-09 18:31:57.763  9854  9854 E AndroidRuntime:        at t7.a.deserialize(Unknown Source:6)
12-09 18:31:57.763  9854  9854 E AndroidRuntime:        at u7.j.i(:11)
12-09 18:31:57.763  9854  9854 E AndroidRuntime:        at v7.s.i(Unknown Source:5)
12-09 18:31:57.763  9854  9854 E AndroidRuntime:        at s7.a.o(:1)
12-09 18:31:57.763  9854  9854 E AndroidRuntime:        at com.sapuseven.untis.models.untis.response.SchoolSearchResult$$serializer.deserialize(Unknown Source:63)
12-09 18:31:57.763  9854  9854 E AndroidRuntime:        at com.sapuseven.untis.models.untis.response.SchoolSearchResult$$serializer.deserialize(Unknown Source:0)
12-09 18:31:57.763  9854  9854 E AndroidRuntime:        at u7.j.i(:11)
12-09 18:31:57.763  9854  9854 E AndroidRuntime:        at v7.s.i(Unknown Source:5)
12-09 18:31:57.763  9854  9854 E AndroidRuntime:        at s7.a.E(:1)
12-09 18:31:57.763  9854  9854 E AndroidRuntime:        at com.sapuseven.untis.models.untis.response.SchoolSearchResponse$$serializer.deserialize(Unknown Source:75)
12-09 18:31:57.763  9854  9854 E AndroidRuntime:        at com.sapuseven.untis.models.untis.response.SchoolSearchResponse$$serializer.deserialize(Unknown Source:0)
12-09 18:31:57.763  9854  9854 E AndroidRuntime:        at u7.j.i(:11)
12-09 18:31:57.763  9854  9854 E AndroidRuntime:        at v7.s.i(Unknown Source:5)
12-09 18:31:57.763  9854  9854 E AndroidRuntime:        at u7.a.b(Unknown Source:21)
12-09 18:31:57.763  9854  9854 E AndroidRuntime:        at m3.r.e(:27)
12-09 18:31:57.763  9854  9854 E AndroidRuntime:        at r4.a.l(:4)
12-09 18:31:57.763  9854  9854 E AndroidRuntime:        at k7.f0.run(:2)
12-09 18:31:57.763  9854  9854 E AndroidRuntime:        at android.os.Handler.handleCallback(Handler.java:938)
12-09 18:31:57.763  9854  9854 E AndroidRuntime:        at android.os.Handler.dispatchMessage(Handler.java:99)
12-09 18:31:57.763  9854  9854 E AndroidRuntime:        at android.os.Looper.loop(Looper.java:233)
12-09 18:31:57.763  9854  9854 E AndroidRuntime:        at android.app.ActivityThread.main(ActivityThread.java:8030)
12-09 18:31:57.763  9854  9854 E AndroidRuntime:        at java.lang.reflect.Method.invoke(Native Method)
12-09 18:31:57.763  9854  9854 E AndroidRuntime:        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:631)
12-09 18:31:57.763  9854  9854 E AndroidRuntime:        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:978)
```
</details>

**Additional information**

 - Android version: _11_
 - BetterUntis version: _3.4.6_
 - Installation source: _SapuSeven.com_

**Comment by SapuSeven** [comment]
Thanks. Looks like an easy fix, so you can expect an update in the next few days.

**Comment by SapuSeven** [comment]
Fixed in 4877f12b62d284af6b5243326d6f4efceed61aa6
